src/Compilers/CSharp/Test/Semantic/Semantics/RefEscapingTests.cs

```diff
@@ -2494,6 +2494,211 @@ .maxstack  1
+        global = local2; // error 2
+    }
+    public static void Main()
+    {
+    }
+}
+public ref struct S
+{
+    public static implicit operator S(in Span<int> s) => throw null;
+    public static S operator ++(in S s) => throw null;
+    public static bool operator ==(in S s1, in Span<int> s2) => throw null;
+    public static bool operator !=(in S s1, in Span<int> s2) => throw null;
+    public override bool Equals(object o) => throw null;
+    public override int GetHashCode() => throw null;
+    public static S operator +(in S s1, in S s2) => throw null;
+}
+";
+            CreateCompilationWithMscorlibAndSpan(text).VerifyDiagnostics(
+                // (9,18): error CS8352: Cannot use local 'local1' in this context because it may expose referenced variables outside of their declaration scope
+                //         global = local1++; // error 1
+                Diagnostic(ErrorCode.ERR_EscapeLocal, "local1++").WithArguments("local1").WithLocation(9, 18),
+                // (12,18): error CS8352: Cannot use local 'local2' in this context because it may expose referenced variables outside of their declaration scope
+                //         global = local2; // error 2
+                Diagnostic(ErrorCode.ERR_EscapeLocal, "local2").WithArguments("local2").WithLocation(12, 18)
+            );
+        }
+
+        [Fact]
+        public void TestMiscInOperators2()
+        {
+            var text = @"
+using System;
+
+public class C
+{
+    public void M(ref S global)
+    {
+        S local1 = stackalloc int[10];
+        local1++;
+
+        Console.Write(global == local1);
+
+        global += local1;
+        local1 += global;
+    }
+    public static void Main()
+    {
+    }
+}
+public ref struct S
+{
+    public static implicit operator S(in Span<int> s) => throw null;
+    public static S operator ++(in S s) => throw null;
+    public static bool operator ==(in S s1, in S s2) => throw null;
+    public static bool operator !=(in S s1, in S s2) => throw null;
+    public override bool Equals(object o) => throw null;
+    public override int GetHashCode() => throw null;
+    public static S operator +(in S s1, in S s2) => throw null;
+}
+";
+            var comp = CreateCompilationWithMscorlibAndSpan(text);
+            comp.VerifyDiagnostics();
+            CompileAndVerify(comp, verify: false).VerifyIL("C.M", @"
+{
+  // Code size       71 (0x47)
+  .maxstack  3
+  .locals init (S V_0, //local1
+                int V_1,
+                System.Span<int> V_2,
+                S V_3)
+  IL_0000:  ldc.i4.s   10
+  IL_0002:  stloc.1
+  IL_0003:  ldloc.1
+  IL_0004:  conv.u
+  IL_0005:  ldc.i4.4
+  IL_0006:  mul.ovf.un
+  IL_0007:  localloc
+  IL_0009:  ldloc.1
+  IL_000a:  newobj     ""System.Span<int>..ctor(void*, int)""
+  IL_000f:  stloc.2
+  IL_0010:  ldloca.s   V_2
+  IL_0012:  call       ""S S.op_Implicit(in System.Span<int>)""
+  IL_0017:  stloc.0
+  IL_0018:  ldloc.0
+  IL_0019:  stloc.3
+  IL_001a:  ldloca.s   V_3
```

Comment:
@VSadov I'm not sure why we create V_0 and use its address (instead of simply using the address of V_3). Seems like a possible optimization.